core/instance/GridBasedInterestManagement.js

```diff
@@ -0,0 +1,280 @@
+        const cellMaxY = (Math.floor(worldMaxY / this.gridCellSize)).clamp(0, this.maxGridYIndex)
+
+        let entitiesInArea = [];
+
+        for (let cellY = cellMinY; cellY <= cellMaxY; cellY++) {
+            for (let cellX = cellMinX; cellX <= cellMaxX; cellX++) {
+
+                const cell = this.getCellFromCellPos(cellX, cellY)
+                for (let s = 0; s < cell.numberOfStaticEntities; s++) {
+                    let entity = cell.staticEntities[s]
+
+                    // this.AABBComparisonsThisTick++
+                    if (entity.x <= worldMaxX
+                        && entity.x >= worldMinX
+                        && entity.y <= worldMaxY
+                        && entity.y >= worldMinY) {
+                        entitiesInArea.push(entity)
+                    }
+                }
+
+                for (let d = 0; d < cell.numberOfDynamicEntities; d++) {
+                    let entity = cell.dynamicEntities[d]
+                    // this.AABBComparisonsThisTick++
+                    if (entity.x <= worldMaxX
+                        && entity.x >= worldMinX
+                        && entity.y <= worldMaxY
+                        && entity.y >= worldMinY) {
+                        entitiesInArea.push(entity)
+                    }
+                }
+            }
+        }
+
+        for (var i = 0; i < this.numberOfBigEntities; i++) {
+            var entity = this.bigEntityList[i]
+            const radius = entity.visibilityRadius || 0
+
+            if (entity.x - radius <= worldMaxX
+                && entity.x + radius >= worldMinX
+                && entity.y - radius <= worldMaxY
+                && entity.y + radius >= worldMinY) {
+                entitiesInArea.push(entity)
+            }
+        }
+
+        // TODO maybe never: gridify events too; for now use identical to old
+        var events = this.events.toArray()
+        let eventsInArea = []
+        for (var i = 0; i < events.length; i++) {
+            var event = events[i]
+            if (event.x <= worldMaxX
+                && event.x >= worldMinX
+                && event.y <= worldMaxY
+                && event.y >= worldMinY) {
+                eventsInArea.push(event)
+            }
+        }
+
+        return { entities: entitiesInArea, events: eventsInArea };
+    }
+
+    update() {
+        // this.AABBComparisonsThisTick = 0
+        // this.numberOfEntitiesNeedingToBeMovedToANewCellThisTick = 0
+        for (let i = 0; i < this.numberOfEntitiesInMasterList; i++) {
+
+            let entity = this.masterEntityList[i]
+
+            // static entities do not move and will never need to have their cell updated
+            if (entity.isStaticEntity) continue
+
+            // Big entities dont exist in cells, skip
+            if (entity.visibilityRadius !== undefined && entity.visibilityRadius >= this.thresholdToBeConsideredABigEntity) continue
```

Comment:
future opt: could probably do both of these checks not during the update (ie don't put static or large entities in the list that update iterates over)